Login.java

```diff
@@ -0,0 +1,354 @@
+
+// March 2017 Version
+
+package com.project.ScafFold;
+
+import java.sql.Timestamp;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.text.SimpleDateFormat;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.google.analytics.tracking.android.EasyTracker;
+
+import android.annotation.SuppressLint;
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.ConnectivityManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.StrictMode;
+import android.view.KeyEvent;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+ 
+
+/*
+ * This is the launching activity of the application
+ * If the user has not logged in then they will be required to register and/or login and then taken to the main activity
+ * If the user is already logged in then they will be directly taken to the main interface
+ */
+
+public class Login extends Activity //Login is an activity
+{
+		final Context context = this; //Use this variable to refer to this activity		
+		
+	    //Variables for use to fetch Shared preferences		
+		public static final String PREFS_NAME = "Prefs"; 
+	    private SharedPreferences settings;
+	    SharedPreferences.Editor editor;
+	   
+	    boolean alreadyRegistered; //check if the user is already registered 
+	    public static final String TAG = "LOGIN SCREEN LOGS";   //Log Tag
+		
+		//Declaring component variables
+	 	Button btnLogin;
+	    Button btnLinkToRegister;
+	    EditText inputEmail;
+	    EditText inputPassword;
+	    TextView ErrorMsg;
+	    
+	    boolean Advanced; //Used to determine if after login, the user proceeds to the advanced interface
+	 
+	    // JSON Response node names
+	    private static String KEY_SUCCESS = "success";
+	    private static String KEY_ERROR = "error";
+	    private static String KEY_EMAIL = "email";
+	    private static String KEY_CREATED_AT = "created_at";	
+	    
+	    //Object to access JSon class for fetching JSON data from the PhP files on server
+	    JSONObject json; 
+	    
+	    //Object to access UserFunctions class that contains the methods for JSON Communication
+	    UserFunctions userFunction;   
+	     
+	    //Variables for Time and Date
+	    SimpleDateFormat dateFormat;
+	  	NumberFormat nf;
+	  	long time;

```

Comment:
Use short because...
```suggestion
	  	short time;

```

Reply:
Thanks. This is done.